num/dualcmplx/dual_example_test.go

```diff
@@ -0,0 +1,134 @@
+// Copyright ©2018 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package dualcmplx_test
+
+import (
+	"fmt"
+	"math"
+	"math/cmplx"
+
+	"gonum.org/v1/gonum/floats"
+	"gonum.org/v1/gonum/num/dualcmplx"
+)
+
+// point is a 2-dimensional point/vector.
+type point struct {
+	x, y float64
+}
+
+// raise raises the dimensionality of a point to a complex.
+func raise(p point) complex128 {
+	return complex(p.x, p.y)
+}
+
+// raiseDual raises the dimensionality of a point to a dual complex number.
+func raiseDual(p point) dualcmplx.Number {
+	return dualcmplx.Number{
+		Real: 1,
+		Dual: complex(p.x, p.y),
+	}
+}
+
+// transform performs the transformation of p by the given set of dual
+// complex transforms in order. The rotations are normalized to unit
+// vectors.
+func transform(p point, by ...dualcmplx.Number) point {
+	if len(by) == 0 {
+		return p
+	}
+
+	// Ensure the modulus of by is correctly scaled.
+	for i := range by {
```

Comment:
Doesn't this negate any dilations?

Reply:
Yes, intentionally. In general cases dilation may be warranted, I didn't want it here.

Reply:
Ok, in that case I'd change 'rotations' to either 'transformations' or some combination or rotations and translations, as this does allow translations. You could mention this is actualy SE(2) as in the paper, which is standard notation in group/ring theory.

Reply:
I need to remember to make the same change to dualquat.